### PR TITLE
Updated all the OpenTK references to use NuGet

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Updated all the OpenTK references to use NuGet
 - Using the same NuGet version of OpenTK across the board
 - Added a gallery factory to allow for platform-specific pages
 - Added the OpenGL galleries to Android, iOS, GTK# and WPF

This is working on all the platforms, to some extent - so I created a nice table to show the details. This is the current state of OpenGL in Xamarin.Forms:

|                      |   Android  |      iOS    |    Linux    |      macOS    |     Windows    |     Tizen        |
|------------:|:----------:|:--------:|:---------:|:------------:|:-------------:|:------------:|
| **Android** |   ✅Both   |                |                  |                      |                        |                      |
|    **GTK#** |                   |                | ⚠️Simple |   ⚠️Simple  |       ✅Both    |                      |
|        **iOS** |                   | ✅Both |                   |                      |                        |                      |
|  **macOS** |                   |               |                   | 🚫Not Impl |                        |                      |
|     **Tizen** |                   |               |                   |                      |                        | 🚫Not Impl |
|      **WPF** |                   |               |                   |                      |      ✅Both     |                      |
|      **UWP** |                   |               |                   |                      |   🚫Not Impl  |                     |

**Notes**
 - Blank cells mean not applicable _(ie: GTK# will never run on iOS - unless some crazy person is busy porting that now...)_
 - Android, iOS and Windows desktop (WPF/GTK#) are all fine
 - GTK# on Linux and macOS seem to work, but the advanced sample crashes creating the shaders.
 - Native macOS and Tizen are not yet implemented
 - UWP is not implemented, but will have to make use of ANGLE so is a lot more work

I checked, and the GTK# Linux and macOS sample were not working on master, so I don't think this is a result of the update. And, they simple sample works, so I am even more convinced. 

### Issues Resolved ### 

- Annoying case where OpenTK is part bundled, par depended on, part mismatched versions.
- Updates to the latest 3.0.1 for the core package, potentially opening up the compatibility for Tizen 5 and it's fork of OpenTK 3.x

### API Changes ###
 
- Updated GTK and WPF to use the latest OpenTK NuGet v3.0.1
- Updated the nuspecs to depend on the NuGet as opposed to bundling

### Platforms Affected ### 

- WPF
- GTK

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

None

### Testing Procedure ###

1. Open the gallery
2. Scroll to the bottom of the samples list
3. Select the OpenGL samples
4. Enjoy!

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by mattleibow** [comment]
Not all the samples work everywhere. The Advanced sample works on all the platforms except Tizen and UWP - as those platforms do not have OpenGL support yet. The Basic sample only runs on the desktop as it requires full OpenGL and not GLES.

**Comment by PureWeen** [comment]
build --uitests

**Comment by mattleibow** [comment]
All the Android tests failed, but I think they are not related:
 - Date Picker Cancel Should Unfocus (Issues.Bugzilla41424.DatePickerCancelShouldUnfocus)
 - Forcing Restart Does Not Cause Crash (Issues.RestartAppTest.ForcingRestartDoesNotCauseCrash)
 - Does Not Crash (Issues.Bugzilla39821.DoesNotCrash)

File: Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj [hunk 1]
```diff
@@ -56,12 +57,13 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <NoWarn>
     </NoWarn>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (paymicro) [inline Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj]: This is necessary? 
I found only one `unsafe` block in XF rero in `GTK` gallery
https://github.com/xamarin/Xamarin.Forms/blob/bd31e1e9fc8b2f9ad94cc99e0c7ab058174821f3/Xamarin.Forms.ControlGallery.GTK/AdvancedOpenGLView.cs#L170

Comment (mattleibow) [reply Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj]: I think so. That file is also used (file linking) in iOS and Android projects. OpenGL for the world, baby! 🌍

**Pull request merged**